src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -2,21 +2,39 @@
 
 import java.util.Optional;
 import mate.academy.hibernate.relations.dao.ActorDao;
+import mate.academy.hibernate.relations.dao.exception.DataProcessingException;
 import mate.academy.hibernate.relations.model.Actor;
+import org.hibernate.Session;
 import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
 
 public class ActorDaoImpl extends AbstractDao implements ActorDao {
     public ActorDaoImpl(SessionFactory sessionFactory) {
         super(sessionFactory);
     }
 
     @Override
-    public Actor add(Actor actor) {
-        return null;
+    public Actor add(Actor actor) throws DataProcessingException {
+        Transaction transaction = null;
+        try (Session session = super.factory.openSession()) {
```

Comment:
When you use try with resource rollback it will not work because the session will be closed before that

Reply:
Please, don't use try with resources when you rollback the transaction, it doesn't work.